packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -709,3 +714,72 @@ class _CupertinoEdgeShadowPainter extends BoxPainter {
     canvas.drawRect(rect, paint);
   }
 }
+
+/// Displays an iOS-style dialog above the current contents of the app.
+///
+/// This function takes a `builder` which typically builds a [CupertinoDialog]
+/// or [CupertinoAlertDialog] widget. Content below the dialog is dimmed with a
+/// [ModalBarrier]. This widget does not share a context with the location that
+/// `showCupertinoDialog` is originally called from. Use a [StatefulBuilder] or
+///  a custom [StatefulWidget] if the dialog needs to update dynamically.
+///
+/// The `context` argument is used to look up the [Navigator] for the dialog.
+/// It is only used when the method is called. Its corresponding widget can
+/// be safely removed from the tree before the dialog is closed.
+///
+/// Returns a [Future] that resolves to the value (if any) that was passed to
+/// [Navigator.pop] when the dialog was closed.
+///
+/// The dialog route created by this method is pushed to the root navigator.
+/// If the application has multiple [Navigator] objects, it may be necessary to
+/// call `Navigator.of(context, rootNavigator: true).pop(result)` to close the
+/// dialog rather than just 'Navigator.pop(context, result)`.
+///
+/// See also:
+///  * [CupertinoDialog], an iOS-style dialog.
+///  * [CupertinoAlertDialog], an iOS-style alert dialog.
+///  * <https://developer.apple.com/ios/human-interface-guidelines/views/alerts/>
+Future<T> showCupertinoDialog<T>({
+  @required BuildContext context,
+  @required WidgetBuilder builder,
+}) {
+  assert(builder != null);
+  return showGeneralDialog(
+    context: context,
+    barrierDismissible: false,
+    barrierColor: _kModalBarrierColor,
+    transitionDuration: const Duration(milliseconds: 300),
+    pageBuilder: (BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation) {
+      return new SafeArea(
+        child: builder(context),
+      );
+    },
+    transitionBuilder: (BuildContext context, Animation<double> animation, Animation<double> secondaryAnimation, Widget child) {
```

Comment:
This function is constant right? you can just declare this statically as a private function in this file instead of instantiating a new instance each time showCupertinoDialog is invoked.